src/encoded/static/components/experiment.js

```diff
@@ -302,6 +302,46 @@ LibrarySubmitterComments.propTypes = {
 };
 
 
+/**
+ * Display, as one element of a "key-value" definition list, all the `construction_platform`
+ * properties in a dataset's replicates' library.
+ */
+const DatasetConstructionPlatform = ({ context }) => {
+    // Collect the library construction platforms from all the dataset replicates.
+    const constructionPlatforms = context.replicates.reduce((platforms, replicate) => {
+        // We easily have duplicates, so only add the replicate library platform if we haven't
+        // already seen it.
+        if ((replicate.library && replicate.library.construction_platform) &&
+            (platforms.findIndex(platform => platform['@id'] === replicate.library.construction_platform['@id']) === -1)) {
+            return platforms.concat(replicate.library.construction_platform);
+        }
+        return platforms;
+    }, []);
+
+    if (constructionPlatforms.length > 0) {
+        return (
+            <div data-test="constructionplatform">
+                <dt>Library construction platform</dt>
+                <dd>
+                    {constructionPlatforms.map((platform, i) =>
+                        <React.Fragment key={platform['@id']}>
```

Comment:
Most likely I am wrong; but that has never deterred me from asking... My understanding of React fragments is that it basically tell react now to complain that there no container-tag. That is why you can use <></>. If that is true, why is there a key in the fragment?

Reply:
You need the key here also, or else you get (and I just tried it removing that key to be sure):

<img width="817" alt="image" src="https://user-images.githubusercontent.com/1181324/75287756-d2850080-57cf-11ea-9981-ee9b1cf01f19.png">